### DISCUSSION TITLE
Forms library and server-side validation

### DISCUSSION THREAD

**Discussion by ataylor32** [primefaces/community#3219] [PrimeVue]
I'm trying to use the PrimeVue forms library, but I don't know how to incorporate server-side validation. I know there is an emphasis on client-side validation, but sometimes error messages _must_ come from the server. For example, some websites are set up so that when you change your password, you're not allowed to change it to one that you've used recently.

My question is: when client-side validation passes, so you submit the form to the server, but then the server says that one or more of the fields have errors, how should I get those fields' errors to display to the user?

**Comment by rammedbro** [comment]
Hi, same here, I have zod both on client and server and i need a way to set zod compatible errors to my primevue form with zod resolver. I think we need api for it, it's pretty common case to set errors or possibly entire state from the outside. Thanks.

**Comment by simonmargesin** [comment]
Hey, 
Not sure if you have found a fix, but i have implemented my own solution for this problem. it requires some hacky code, since primevue doesn't expose any official endpoints for setting field errors programatically, like setFieldError.

What i do, is after my api call rejects, i pass the form template ref and the server response to a function, where i set the field error manually. This will display an error message beneath the field, which will dissapear just like any frontend validation on blur or field change.

I am aware the Code is a bit of a mess, but i had to cut it out of my application and gut it for general repurposing:

https://stackblitz.com/edit/ucwvzcnq-gv1jzjzp?file=src%2FCustomValidation.js
